src/acl_usm.cpp

```diff
@@ -143,12 +143,14 @@ CL_API_ENTRY void *CL_API_CALL clHostMemAllocINTEL(
     }
   }
 
+  bool track_mem_id = false;
   if (acl_get_hal()->host_alloc) {
     std::array<mem_properties_t, 3> mmd_properties;
     {
       auto mmd_properties_it = mmd_properties.begin();
       if (mem_id) {
-        if (acl_platform.offline_mode == ACL_CONTEXT_MPSIM) {
+        if (acl_get_hal()->support_buffer_location(devices)) {
```

Comment:
Will this break backward compatibility on simulation device prior to 2024.2?

Reply:
Note: since BL is a SYCL/oneAPI feature instead of the OpenCL feature and we don't have hard backward compatibility requirement for oneAPI. If there is no way to keep the backward compatibility of this, I think this is OK to break it.

Reply:
> Will this break backward compatibility on simulation device prior to 2024.2? E.g., new runtime will not work with the 2024.1 compiled exe for buffer locations.

Yes it will break backwards compatibility, I talked with Joe and the expectation is that users will use compatible simulation MMD and the runtime since they are bundled together in oneAPI.

Reply:
Sounds good!